extensions/Corbnorb/noise.js

```diff
@@ -0,0 +1,3520 @@
+              },
+              X: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+              Y: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+              Z: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+              INVERTED: {
+                type: ArgumentType.STRING,
+                menu: "INVERTED_MENU",
+                defaultValue: "false",
+              },
+              EASING: {
+                type: ArgumentType.STRING,
+                menu: "EASING_TYPE",
+                defaultValue: "Linear",
+              },
+            },
+          },
+        ],
+        menus: {
+          NOISES: {
+            acceptReporters: true,
+            items: "getNoises",
+          },
+          NOISE_TYPE: {
+            acceptReporters: true,
+            items: [
+              "openSimpex2",
+              "openSimpex2S",
+              "cellular",
+              "perlin",
+              "value Cubic",
+              "value",
+            ],
+          },
+          FRACTAL_TYPE: {
+            acceptReporters: true,
+            items: ["none", "fbm", "ridged", "ping Pong"],
+          },
+          INVERTED_MENU: {
+            acceptReporters: true,
+            items: [
+              { text: "true", value: "true" },
+              { text: "false", value: "false" },
+            ],
+          },
+          EASING_TYPE: {
+            acceptReporters: true,
+            items: ["linear", "squared", "cubed", "root"],
+          },
+        },
+      };
+    }
+
+    getNoises() {
+      return noiseMenu;
+    }
+
+    initNoise(args) {
+      const id = Cast.toString(args.ID);
+      const seed = Cast.toNumber(args.SEED);
+      const fractal = Cast.toString(args.FRACTAL);
+      const frequency = Cast.toNumber(args.FREQUENCY);
+      const octaves = Cast.toNumber(args.OCTAVES);
+      noises[id] = new FastNoiseLite(seed);
+      if (!noiseMenu.some((e) => e.value === id)) {
+        noiseMenu.push({ text: id, value: id });
+      }
+      switch (args.TYPE) {
+        case "openSimplex2":
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.OpenSimplex2);
+          break;
+        case "openSimplex2S":
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.OpenSimplex2S);
+          break;
+        case "cellular":
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.Cellular);
+          break;
+        case "perlin":
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.Perlin);
+          break;
+        case "value cubic":
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.ValueCubic);
+          break;
+        case "value":
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.Value);
+          break;
+        default:
+          noises[id].SetNoiseType(FastNoiseLite.NoiseType.Perlin);
+          break;
+      }
```

Comment:
You could simplify this by renaming your argument values and doing something like:

```suggestion
    noises[id].SetNoiseType(FastNoiseLite.NoiseType[args.TYPE])
```
instead of having the whole switch block
You could also do something similar for the one below